init.go

```diff
@@ -39,7 +42,16 @@ func Run(m *testing.M, opts ...agent.Option) int {
 		defer scopetesting.UnpatchTestingLogger()
 	}
 
-	defer newAgent.Stop()
+	// Handle SIGINT and SIGTERM
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sigs
+		fmt.Println("Terminating agent, sending partial results...")
+		newAgent.Stop()
+		os.Exit(1)
+	}()
```

Comment:
I'm not sure about this. Doesn't the `testing` library handle these signals already?

Reply:
Well I think the handle is from the go test command but not the `testing` library.

**Pre Notify:** Partials unsent results are lost...
<img width="378" alt="image" src="https://user-images.githubusercontent.com/69803/74197934-a8283480-4c60-11ea-94e6-909fa2bf612e.png">

**Post Notify:**  We ensure partials results are sent before dying...
<img width="610" alt="image" src="https://user-images.githubusercontent.com/69803/74198062-e3c2fe80-4c60-11ea-9c4d-0a7f4048df97.png">